pkg/ttnpb/fieldmask_utils.go

```diff
@@ -267,7 +267,7 @@ func FieldsWithPrefix(prefix string, paths ...string) []string {
 func FieldsWithoutPrefix(prefix string, paths ...string) []string {
 	ret := make([]string, 0, len(paths))
 	for _, p := range paths {
-		if i := strings.Index(p, prefix+"."); i >= 0 {
+		if strings.HasPrefix(p, prefix+".") {
```

Comment:
The index of `mac_state.` in `pending_mac_state.` is greater or equal to 0 here, but this is obviously not the intended behavior.